lib/AdapterRpc.php

```diff
@@ -154,7 +156,45 @@ public function getUserAttributes($user, $attrNames)
 
 	public  function isUserOnFacility($spEntityId, $userId)
 	{
-		throw new BadMethodCallException("NotImplementedException");
+		$facilities = sspmod_perun_RpcConnector::get('facilitiesManager', 'getFacilitiesByAttribute', array(
+			'attributeName' => 'urn:perun:facility:attribute-def:def:entityID',
+			'attributeValue' => $spEntityId,
+		));
+
+		$allowedResources = array();
+		foreach ($facilities as $facility) {
+			$resources = sspmod_perun_RpcConnector::get('facilitiesManager', 'getAssignedResources', array(
+				'facility' => $facility['id'],
+			));
+			$allowedResources = array_merge($allowedResources, $resources);
+		}
+
+		$members = sspmod_perun_RpcConnector::get('membersManager', 'getMembersByUser', array(
+			'user' => $userId,
+		));
+
+		$validMembers = array();
+		foreach ($members as $member) {
+			if ($member['status'] === 'VALID') {
+				array_push($validMembers, $member);
+			}
+		}
+
+		$allGroups = array();
+		foreach ($allowedResources as $resource) {
+			foreach ($validMembers as $member) {
+				$groups = sspmod_perun_RpcConnector::get('resourcesManager', 'getAssignedGroups', array(
+					'resource' => $resource['id'],
+					'member' => $member['id'],
+				));
+				foreach ($groups as $group) {
+					array_push($allGroups, new sspmod_perun_model_Group($group['id'], $group['name'], $group['description']));
+				}
+			}
+		}
+
+		$allGroups = $this->removeDuplicateEntities($allGroups);
+		return $allGroups;
```

Comment:
Shouldn't method isUserOnFacility return just true/false? Otherwise the method should have the name getUsersGroupsOnFacility or similarly.

Reply:
No, this method returns array of groups, because we put this list of groups into request for later processing in other modules. We can rename this method to getUsersGroupsOnFacility or similar. 

I will prepare new commit with renaming this method and all calls.